### PR TITLE
Add some vendor telemetry data for each firmware file

### PR DESCRIPTION
**Pull request by hughsie**
Use this to update the database:

   ALTER TABLE firmware ADD download_cnt INT DEFAULT 0;
   CREATE INDEX filename_idx ON clients (filename(40));

Fixes https://github.com/hughsie/lvfs-website/issues/32

### CODE REVIEW

File: app/templates/telemetry.html [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+<div class="container mt-3">
+
+<h2>Telemetry for group {{group_id}}</h2>
+<div class="container mt-3">
+<table class="table">
+  <tr class="row">
+    <th class="col-sm-4">Hardware</th>
+    <th class="col-sm-2">
+      Version
+      <a href="/lvfs/telemetry/{{age}}/nameversion/up">▼</a>
```
Comment (superm1) [inline app/templates/telemetry.html]: If possible, can you make it clearer which way the page is currently sorted?  As it stands, the page renders the same no matter what sort you pick.

File: app/templates/telemetry.html [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+{% endfor %}
+  <tr class="row">
+    <td class="col-sm-4"><b>&nbsp;</b></td>
+    <td class="col-sm-2"><b>&nbsp;</b></td>
+    <td class="col-sm-2"><b>{{total_downloads}}</b></td>
+    <td class="col-sm-2"><b>{{total_success}}</b></td>
+    <td class="col-sm-2"><b>{{total_failed}}</b></td>
+  </tr>
+</table>
+{% if show_duplicate_warning %}
```
Comment (superm1) [inline app/templates/telemetry.html]: Can you move this to the top of the page instead of the bottom?

File: app/views_firmware.py [hunk 3]
```diff
@@ -436,6 +453,113 @@ def firmware_component_show(firmware_id, cid, page='overview'):
...
+            continue
+
+        # reports
+        if age == 0:
+            cnt_download = fw.download_cnt
+        else:
+            cnt_download = db.clients.get_firmware_count_filename(fw.filename, age)
+        cnt_success = 0
+        cnt_failed = 0
+        rpts = db.reports.get_all_for_firmware_id(fw.firmware_id, age=age, limit=999)
```
Comment (superm1) [inline app/views_firmware.py]: Why is this limited to 999?

File: app/templates/telemetry.html (hunk 1 above)
Comment (hughsie) [reply app/templates/telemetry.html]: Fixed, now one of the arrows is always orange.

File: app/templates/telemetry.html (hunk 2 above)
Comment (hughsie) [reply app/templates/telemetry.html]: Fixed.

File: app/views_firmware.py (hunk 3 above)
Comment (hughsie) [reply app/views_firmware.py]: It was a hack. Fixed properly now.

**Pull request merged**